src/Common/src/Interop/Windows/Winsock/Interop.WinsockAsync.cs

```diff
@@ -161,14 +165,29 @@ internal struct NetworkEvents
         }
 
         [StructLayout(LayoutKind.Sequential)]
-        internal struct WSAMsg
+        internal unsafe struct WSAMsg
         {
-            internal IntPtr socketAddress;
+            internal byte* socketAddress;
             internal uint addressLength;
-            internal IntPtr buffers;
+            internal WSABuffer* buffers;
             internal uint count;
             internal WSABuffer controlBuffer;
             internal SocketFlags flags;
+            internal int pad;
+
+            // NOTE: Even though Win32 docs say the flags arg is a DWORD, and thus 4 bytes long, it seems to actually write 4 bytes past this.
+            // The "pad" field above deals with this.
```

Comment:
NOTE: A quick check shows that explicitly declaring the pad is unnecessary as `Marshal.SizeOf<WSAMSG>` is correctly reporting the expected sizes. I validated this using declarations as:
```C#
[StructLayout(LayoutKind.Sequential)]
struct WSABUF
{
    public IntPtr len;
    public IntPtr buf;
}

[StructLayout(LayoutKind.Sequential)]
struct WSAMSG
{
    public IntPtr name;
    public int namelen;
    public IntPtr lpBuffers;
    public uint dwBufferCount;
    public WSABUF Control;
    public uint dwFlags;
}
```